lib/web_ui/lib/src/engine/text_editing/text_capitalization.dart

```diff
@@ -0,0 +1,92 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of engine;
+
+/// Controls the capitalization of the text.
+///
+/// This corresponds to Flutter's [TextCapitalization].
+///
+/// Uses `text-transform` css property.
+/// See: https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform
+enum TextCapitalization {
+  /// Uppercase for the first letter of each word.
+  words,
+
+  /// Currently not implemented on Flutter Web. Uppercase for the first letter
+  /// of each sentence.
+  sentences,
+
+  /// Uppercase for each letter.
+  characters,
+
+  /// Lowercase for each letter.
+  none,
+}
+
+/// Helper class for text capitalization.
+///
+/// Uses `text-transform` css property.
+/// See: https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform
+class TextCapitalizationConfig {
+  final TextCapitalization textCapitalization;
+
+  static final RegExp wordExp = new RegExp(r"(\w+)");
+  static final RegExp whiteSpaceExp = new RegExp(r"(\s+)");
+
+  const TextCapitalizationConfig.defaultCapitalization()
+      : textCapitalization = TextCapitalization.none;
+
+  // TODO: support sentence level text capitalization.
+  TextCapitalizationConfig.fromInputConfiguration(String inputConfiguration)
+      : this.textCapitalization =
+            inputConfiguration == 'TextCapitalization.words'
+                ? TextCapitalization.words
+                : (inputConfiguration == 'TextCapitalization.characters')
+                    ? TextCapitalization.characters
+                    : TextCapitalization.none;
+
+  /// Sets `autocapitalize` attribute on input elements.
+  ///
+  /// This attribute is only available for mobile browsers.
+  ///
+  /// Note that in mobile browsers the onscreen keyboards provide sentence
+  /// level capitalization as default as apposed to no capitalization on desktop
+  /// browser.
+  ///
+  /// See: https://developers.google.com/web/updates/2015/04/autocapitalize
+  /// https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autocapitalize
+  void setAutocapitalizeAttribute(html.HtmlElement domElement) {
+    String autocapitalize = '';
+    switch (textCapitalization) {
+      case TextCapitalization.words:
+        // TODO: There is a bug for `words` level capitalization in IOS now.
+        // For now go back to default. Remove the check after bug is resolved.
+        // https://bugs.webkit.org/show_bug.cgi?id=148504
+        if(browserEngine == BrowserEngine.webkit) {
+           autocapitalize = 'sentences';
+        } else {
+          autocapitalize = 'words';
+        }
+        break;
+      case TextCapitalization.characters:
+        autocapitalize = 'characters';
+        break;
+      case TextCapitalization.sentences:
+        autocapitalize = 'sentences';
+        break;
+      case TextCapitalization.none:
+      default:
+        autocapitalize = 'off';
+        break;
+    }
```

Comment:
Suggestion: I would probably put this in a getter:
```dart
String get autocapitalizeAttribute {
  switch (textCapitalization) {
    case TextCapitalization.words:
      return browserEngine == BrowserEngine.webkit ? 'sentences' : 'words';
    case TextCapitalization.characters:
      return 'characters';
    // ...
  }
  return 'off';
}

void setAutocapitalizeAttribute(html.HtmlElement domElement) {
  if (domElement is html.InputElement) {
    html.InputElement element = domElement;
    element.setAttribute('autocapitalize', autocapitalizeAttribute);
  } else if (domElement is html.TextAreaElement) {
    html.TextAreaElement element = domElement;
    element.setAttribute('autocapitalize', autocapitalizeAttribute);
  }
}
```

This is a personal preference, so it's not a blocker.

Reply:
I think I'll keep this since we converted to this type of attribute setting (applying the attribute inside the class that carries that property) in other features. For example: InputType, Autofillinfo, EditableTextGeometry.